plugin/trino-mysql/src/main/java/io/trino/plugin/mysql/MySqlClient.java

```diff
@@ -314,13 +323,37 @@ protected String getTableSchemaName(ResultSet resultSet)
 
             case Types.TIMESTAMP:
                 // TODO support higher precisions (https://github.com/trinodb/trino/issues/6910)
-                break; // currently handled by the default mappings
+                return Optional.of(ColumnMapping.longMapping(
+                        TIMESTAMP_MILLIS,
+                        timestampReadFunction(TIMESTAMP_MILLIS),
+                        timestampWriteFunction(TIMESTAMP_MILLIS),
+                        DISABLE_PUSHDOWN));
         }
 
         // TODO add explicit mappings
         return legacyColumnMapping(session, connection, typeHandle);
     }
 
+    public static LongReadFunction timestampReadFunction(TimestampType timestampType)
+    {
+        checkArgument(timestampType.getPrecision() <= TimestampType.MAX_SHORT_PRECISION, "Precision is out of range: %s", timestampType.getPrecision());
+        return (resultSet, columnIndex) -> toTrinoTimestamp(timestampType, resultSet.getObject(columnIndex, LocalDateTime.class));
+    }
```

Comment:
is this different from `io.trino.plugin.jdbc.StandardColumnMappings#timestampReadFunction`?

Reply:
Similar, but `toTrinoTimestamp` is implemented locally to support milliseconds rounding

Reply:
@findepi Can we "fix" `StandardColumnMappings#timestampReadFunction` to perform rounding always since that's what every connector seems to want to do. Or introduce a new method similar to `StandardColumnMappings#timestampColumnMappingUsingSqlTimestampWithRounding`?

(As a separate effort).

Reply:
> `StandardColumnMappings#timestampReadFunction` to perform rounding always since that's what every connector seems to want to do.

That requires opt-int, since it should come along with DISABLE_PUSHDOWN.

However, maybe we don't need (at least not it in the long term)? Trino timestamp precision >= everyone else's, so we should be able to map data types without any rounding.